src/main/editDelete/preProcess.ts

```diff
@@ -50,6 +31,82 @@ const injectAttributes: MapCallback<PartialWord, Word> = (word, i) => ({
   fileName: 'PLACEHOLDER FILENAME',
 });
 
+const constructWord = (
+  word: string,
+  startTime: number,
+  duration: number,
+  outputStartTime: number,
+  key: string
+) => {
+  const newWord: Word = {
+    word,
+    startTime: roundToMs(startTime),
+    duration: roundToMs(duration),
+    outputStartTime: roundToMs(outputStartTime),
+    deleted: false,
+    key,
+    fileName: 'PLACEHOLDER FILENAME',
+  };
+
+  return newWord;
+};
+
+/**
+ * Adds spaces between words which represent the silence between each word
+ * @param result The output being constructed
+ * @param word The current element of words
+ * @param index The index of word in words
+ * @param words The list of words being reduced
+ * @returns The updated transcript with a silence after word
+ */
+let TOTAL_DURATION = 0;
+const addSpaces = (
+  result: Word[],
+  word: Word,
+  index: number,
+  words: Word[]
+) => {
+  const spaceChar = ' ';
+
+  // is the first word
+  if (index === 0) {
+    result.push(
+      constructWord(spaceChar, 0, words[index].startTime, 0, index.toString())
+    );
+  }
+  // not the last word
+  if (index < words.length - 1) {
+    word.key = (index * 2 + 1).toString();
+
+    result.push(word);
+    result.push(
+      constructWord(
+        spaceChar,
+        word.startTime + word.duration,
+        words[index + 1].startTime - word.startTime - word.duration,
+        word.startTime + word.duration,
+        (index * 2 + 2).toString()
+      )
+    );
+  }
+  // is last word
+  else {
```

Comment:
It might be better here to use a similar approach to how Patrick did `fillDurationGaps` to avoid duplicating code.

Reply:
good call, addressed it here: https://github.com/patrickbrett/mlvet/pull/54/commits/837cda0465a8fef63d9ed07412c3e779d1507079